packages/files-ui/src/Components/Modules/FileBrowsers/CreateShareModal.tsx

```diff
@@ -0,0 +1,225 @@
+  }
+)
+
+interface ICreateShareModalProps {
+  modalOpen: boolean
+  close: () => void
+}
+
+const CreateShareModal: React.FC<ICreateShareModalProps> = ({
+  modalOpen,
+  close
+}: ICreateShareModalProps) => {
+  const classes = useStyles()
+  const { createShare } = useFiles()
+  const { filesApiClient } = useFilesApi()
+  const [creatingShare, setCreatingShare] = useState(false)
+  const [shareName, setShareName] = useState("")
+  const [shareUsers, setShareUsers] = useState<Array<{label: string; value: LookupUser}>>([])
+  const [permissions, setPermissions] = useState<"read" | "write" | undefined>(undefined)
+  const { desktop } = useThemeSwitcher()
+  const inputRef = useRef<any>(null)
+
+  useEffect(() => {
+    if (modalOpen) {
+      setTimeout(() => inputRef.current?.focus(), 100)
+    }
+  }, [modalOpen])
+
+  const handleLookupUser = useCallback(async (inputVal: string) => {
+    if (inputVal === "") return []
+    const lookupBody: LookupUserRequest = {}
+    const ethAddressRegex = new RegExp("^0x[a-fA-F0-9]{40}$") // Eth Address Starting with 0x and 40 HEX chars
+    const pubKeyRegex = new RegExp("^0x[a-fA-F0-9]{66}$") // Compressed public key, 66 chars long
+
+    if (ethAddressRegex.test(inputVal)) {
+      lookupBody.public_address = inputVal
+    } else if (pubKeyRegex.test(inputVal)) {
+      lookupBody.identity_public_key = inputVal
+    } else {
+      lookupBody.username = inputVal
+    }
+
+    const result = await filesApiClient.lookupUser(lookupBody)
+
+    if (!result) return []
+    const currentUsers = shareUsers.map(su => su.value.uuid)
+    if (currentUsers.includes(result.uuid)) return []
+
+    return [{ label: inputVal, value: result }]
+  }, [filesApiClient, shareUsers])
+
+  const handleCreateShare = useCallback(async () => {
+    const users = shareUsers.map(su => ({ uuid: su.value.uuid, pubKey: EthCrypto.publicKey.decompress(su.value.identity_pubkey.slice(2)) }))
+    const readers = (permissions === "read") ? users : []
+    const writers = (permissions === "write") ? users : []
+    setCreatingShare(true)
+    createShare(shareName, writers, readers)
+      .then(close)
+      .catch(console.error)
+      .finally(() => setCreatingShare(false))
+  }, [shareUsers, createShare, permissions, shareName, close])
+
+  return (
+    <CustomModal
+      className={classes.modalRoot}
+      injectedClass={{
+        inner: classes.modalInner
+      }}
+      active={modalOpen}
+      closePosition="none"
+      maxWidth="sm"
+    >
+      <div className={classes.root}>
+        <div className={classes.iconBacking}>
+          <ShareAltSvg />
+        </div>
+        <div className={classes.heading}>
+          <Typography variant='body1'>
+            <Trans>Share</Trans>
+          </Typography>
+        </div>
+        <div className={classes.modalFlexItem}>
+          <TextInput
+            label='Name'
+            value={shareName}
+            onChange={(value) => {setShareName(value as string)}} />
```

Comment:
toString() would be better
```suggestion
            onChange={(value) => {setShareName(value.toString())}} />
```